contracts/schemes/ContributionRewardExt.sol

```diff
@@ -0,0 +1,454 @@
+    *         rewards[1] - Amount of ETH requested per period
+    *         rewards[2] - Amount of external tokens requested per period
+    * @param _externalToken Address of external token, if reward is requested there
+    * @param _beneficiary Who gets the rewards. if equal to 0 the beneficiary will be msg.sender.
+    * @param _proposer proposer . if equal to 0 the proposer will be msg.sender.
+    */
+    function proposeContributionReward(
+        string memory _descriptionHash,
+        int256 _reputationChange,
+        uint[3] memory _rewards,
+        IERC20 _externalToken,
+        address payable _beneficiary,
+        address _proposer
+    )
+    public
+    returns(bytes32)
+    {
+        address proposer = _proposer;
+        if (proposer == address(0)) {
+            proposer = msg.sender;
+        }
+        bytes32 contributionId = votingMachine.propose(2, voteParams, proposer, address(avatar));
+        address payable beneficiary = _beneficiary;
+        if (beneficiary == address(0)) {
+            beneficiary = msg.sender;
+        }
+
+        ContributionProposal memory proposal = ContributionProposal({
+            nativeTokenReward: _rewards[0],
+            reputationChange: _reputationChange,
+            ethReward: _rewards[1],
+            externalToken: _externalToken,
+            externalTokenReward: _rewards[2],
+            beneficiary: beneficiary,
+            executionTime: 0,
+            nativeTokenRewardLeft: 0,
+            reputationChangeLeft: 0,
+            ethRewardLeft: 0,
+            externalTokenRewardLeft: 0
+        });
+        organizationProposals[contributionId] = proposal;
+
+        emit NewContributionProposal(
+            address(avatar),
+            contributionId,
+            address(votingMachine),
+            _descriptionHash,
+            _reputationChange,
+            _rewards,
+            _externalToken,
+            beneficiary,
+            proposer
+        );
+
+        proposalsInfo[address(votingMachine)][contributionId] = ProposalInfo({
+            blockNumber:block.number,
+            avatar:avatar
+        });
+        return contributionId;
+    }
+
+    /**
+    * @dev RedeemReputation reward for proposal
+    * @param _proposalId the ID of the voting in the voting machine
+    * @return reputation the redeemed reputation.
+    */
+    function redeemReputation(bytes32 _proposalId) public returns(int256 reputation) {
+        ContributionProposal memory _proposal = organizationProposals[_proposalId];
+        ContributionProposal storage proposal = organizationProposals[_proposalId];
+        require(proposal.executionTime != 0);
+        //set proposal reward to zero to prevent reentrancy attack.
+        proposal.reputationChange = 0;
+
+        if (_proposal.beneficiary == address(this)) {
+            if (_proposal.reputationChange > 0) {//for now only mint(not burn) rep allowed from ext contract.
```

Comment:
Why is burn not being allowed?

Reply:
because it is complicated the rewarder contract.and there is no real req re that yet